### PR TITLE
Remove any Kolibri python dependency for building frontend assets

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Previous work has tried to make frontend asset building an exportable, independent library that can be seamlessly imported into external plugin projects without significant overhead.
* This has so far failed, still requiring the `kolibri` package to be pip installed, and even then that not really working most of the time.
* This PR resolves this by removing any dependency on `kolibri` itself when doing frontend asset building.
* It mostly achieves this by putting all build configuration into `webpack.config.js` inside the plugin folder.
* While most of the build time configuration is now stored in Javascript, getting the version of the plugin to use at build time, the file system path for the Python module path specified plugin, and the correct path for locale files is still most easily calculated using Python, so there is still some Python code doing work here.
* This also requires that the plugin (or Kolibri) is locally available for import, in order to infer version information - would like feedback on this approach.
* The WebpackBundleHook now just needs the `unique_slug` which is in common between the Python code and the `webpack.config.js`
* As further cleanup, this PR removes all of the unnecessary boilerplate, and defaults to using the `webpack_asset` template tag to load a specific SPA's main module into its page. We should add the sync hooks/async hooks back in as they are needed and to the extent that we want SPAs to be pluggable.
* Also updates a few kolibri-tools dependencies and makes vue-template-compiler a hard dependency again (but with a more flexible version requirement).

### Reviewer guidance
Does it build?
Does it build right?

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/user/templates/user/user.html [hunk 1]
```diff
@@ -1,8 +1,5 @@
 {% extends "kolibri/base.html" %}
-{% load user_tags %}
-{% load kolibri_tags %}
-{% load cache %}
+{% load webpack_tags %}
 {% block content %}
-  {% user_async_assets %}
-  {% user_assets %}
```
Comment (rtibbles) [inline kolibri/plugins/user/templates/user/user.html]: Need to revert this change for `user_assets` and related hooks, as it is currently being used in the demo_server plugin.

File: packages/kolibri-tools/lib/webpack_json.py [hunk 2]
```diff
@@ -90,53 +53,60 @@ def expand_glob(build_item):
...
-    try:
-        import_string = app + ".kolibri_plugin"
-        if path:
-            # Need to define the base module by its path first
-            import_package(app, os.path.join(path, "__init__.py"))
-            # Then import the plugin module after
-            plugin_module = import_package(
-                import_string, os.path.join(path, "kolibri_plugin.py")
+def plugin_data(module_path):
+    if resource_exists(module_path, "webpack.config.js"):
```
Comment (indirectlylit) [inline packages/kolibri-tools/lib/webpack_json.py]: this line is currently throwing an error for me when running `yarn run devserver`:

```
(kolibri) d@slate ~/P/l/kolibri (independent_build)> yarn run devserver
yarn run v1.15.2
$ npm-run-all --parallel django-devserver lint-frontend:watch:format hashi-dev watch services
npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 django-devserver /Users/d/Projects/le/kolibri
> kolibri --debug manage runserver --settings=kolibri.deployment.default.settings.dev "0.0.0.0:8000" 


> kolibri-root@0.12.0-dev.1 lint-frontend:watch:format /Users/d/Projects/le/kolibri
> yarn run lint-frontend --monitor --write

npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 hashi-dev /Users/d/Projects/le/kolibri
> yarn workspace hashi run dev

npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 watch /Users/d/Projects/le/kolibri
> kolibri-tools build dev --file ./build_tools/build_plugins.txt

npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 services /Users/d/Projects/le/kolibri
> kolibri services --foreground --skipupdate

$ kolibri-tools lint '{kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}' --ignore '**/node_modules/**','**/static/**','**/packages/kolibri-core-for-export/**' --monitor --write
WARNING:root:No C Extensions available for this platform.

WARNING:root:No C Extensions available for this platform.

$ yarn run build-base --mode=development --watch
$ webpack --config ./webpack.config.js --mode=development --watch
Kolibri Linter: Initializing watcher for the following patterns: {kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE

webpack is watching the files…

WARNING:root:No C Extensions available for this platform.

Traceback (most recent call last):
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 177, in <module>
    main()
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 162, in main
    result = initialize_plugins(build_list)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 106, in initialize_plugins
    plugins += expand_glob(build_item)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 60, in expand_glob
    plugins.append(plugin_data(child_module_path))
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 69, in plugin_data
    if resource_exists(module_path, "webpack.config.js"):
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1186, in resource_exists
    return get_provider(package_or_requirement).has_resource(resource_name)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 406, in get_provider
    return _find_adapter(_provider_factories, loader)(module)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1439, in __init__
    self.module_path = os.path.dirname(getattr(module, "__file__", ""))
  File "/Users/d/Projects/le/kolibri/.venv/bin/../lib/python3.7/posixpath.py", line 156, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
child_process.js:656
    throw err;
    ^

Error: Command failed: PATH=$(echo $PATH | sed 's/\/usr\/bin://g')":/usr/bin" python /Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py --output_file /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/11979-79645-i0wsdn.2gc1.json --plugin_file ./build_tools/build_plugins.txt
WARNING:root:No C Extensions available for this platform.

Traceback (most recent call last):
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 177, in <module>
    main()
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 162, in main
    result = initialize_plugins(build_list)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 106, in initialize_plugins
    plugins += expand_glob(build_item)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 60, in expand_glob
    plugins.append(plugin_data(child_module_path))
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 69, in plugin_data
    if resource_exists(module_path, "webpack.config.js"):
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1186, in resource_exists
    return get_provider(package_or_requirement).has_resource(resource_name)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 406, in get_provider
    return _find_adapter(_provider_factories, loader)(module)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1439, in __init__
    self.module_path = os.path.dirname(getattr(module, "__file__", ""))
  File "/Users/d/Projects/le/kolibri/.venv/bin/../lib/python3.7/posixpath.py", line 156, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:653:13)
    at module.exports (/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/read_webpack_json.js:72:5)
    at Command.<anonymous> (/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/cli.js:132:24)
    at Command.listener (/Users/d/Projects/le/kolibri/node_modules/commander/index.js:315:8)
    at Command.emit (events.js:189:13)
    at Command.parseArgs (/Users/d/Projects/le/kolibri/node_modules/commander/index.js:651:12)
    at Command.parse (/Users/d/Projects/le/kolibri/node_modules/commander/index.js:474:21)
    at checkVersion (/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/cli.js:342:13)
    at /Users/d/Projects/le/kolibri/node_modules/check-node-version/index.js:133:7
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kolibri-root@0.12.0-dev.1 watch: `kolibri-tools build dev --file ./build_tools/build_plugins.txt`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/d/.npm/_logs/2019-08-09T16_55_08_135Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kolibri-root@0.12.0-dev.1 hashi-dev: `yarn workspace hashi run dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 hashi-dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! kolibri-root@0.12.0-dev.1 lint-frontend:watch:format: `yarn run lint-frontend --monitor --write`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 lint-frontend:watch:format script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/d/.npm/_logs/2019-08-09T16_55_08_275Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/d/.npm/_logs/2019-08-09T16_55_08_276Z-debug.log
ERROR: "watch" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Comment (rtibbles) [reply packages/kolibri-tools/lib/webpack_json.py]: Hrm, wonder if reading the lines is happening differently on a Mac and it is getting a None somewhere.

Just need to add a `None` catch at the top of the function I think. Updating now.

**Comment by indirectlylit** [comment]
work moved to #5864
